src/libs/memoize/cache/arrayCacheBuilder.ts

```diff
@@ -0,0 +1,74 @@
+import type {Cache, CacheOpts} from '@libs/memoize/types';
+import {getEqualityComparator} from '@libs/memoize/utils';
+
+/**
+ * Builder of the cache using `Array` primitive under the hood.
+ * @param opts - Cache options, check `CacheOpts` type for more details.
+ * @returns
+ */
+function buildArrayCache<K extends unknown[], V>(opts: CacheOpts): Cache<K, V> {
+    const cache: Array<[K, V]> = [];
+
+    const keyComparator = getEqualityComparator(opts);
+
+    function getKeyIndex(key: K) {
+        for (let i = cache.length - 1; i >= 0; i--) {
+            if (keyComparator(cache[i][0], key)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    return {
+        get(key) {
+            const index = getKeyIndex(key);
+
+            if (index !== -1) {
```

Comment:
if it is -1,  should we return `null` or something?

Reply:
It will return `undefined`. Do you think we should change `undefined` to `null` in this case?

Reply:
I think `undefined` or `null` is fine, but let's make it explicit with an early return:

```ts
if (index === -1) {
    return null;
}
const [entry] = cache.splice(index, 1);
cache.push(entry);
return {value: entry[1]};
```